### PR TITLE
Query type

### PR DESCRIPTION
**Pull request by dorisRiver**

### CODE REVIEW

File: api/service/query_type.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+// QueryType handles the /query request.
+func (s *Service) QueryType(c *gin.Context) (*queryResp, error) {
+	value := c.Query("value")
+	if value == "" {
+		return nil, errMissingQueryValue
+	}
+
+	queryType := "block"
+	switch {
+	case regexp.MustCompile(slotRegMatch).MatchString(value):
```
Comment (maxkaneco) [inline api/service/query_type.go:32]: why not compile once in constructor? There is no need to compile the same pattern every time?

File: api/service/query_type.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/photon-storage/go-photon/crypto/bls"
+
+	"github.com/photon-storage/photon-explorer/database/orm"
+)
+
+type queryResp struct {
+	Type string `json:"type"`
```
Comment (maxkaneco) [inline api/service/query_type.go:14]: The concrete data is to be added later on?

Comment (dorisRiver) [reply api/service/query_type.go:14]: this type is to let the front end know which api to call for the concrete data.

**Review by maxkaneco — APPROVED** [review]

**Comment by maxkaneco** [comment]
PPH=1hr

**Pull request merged**